packages/grafana-data/src/types/datasource.ts

```diff
@@ -560,6 +560,7 @@ export interface DataQueryRequest<TQuery extends DataQuery = DataQuery> {
 
   /** Filters to dynamically apply to all queries */
   filters?: AdHocVariableFilter[];
+  groupByDimensions?: string[];
```

Comment:
looks good, only thing to consider is if it should be named `groupByKeys` , to use sameish name as we have for getTagKeys